### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by dims**
Signed-off-by: Davanum Srinivas <davanum@gmail.com>

**What this PR does / why we need it**:
Once we are happy with this this, we can drop the travis based tests.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by neolit123** [comment]
+1 on using using this over travis.
/cc

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Test
+on: [push, pull_request]
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-versions: [1.12.x, 1.13.x, 1.14.x]
+        platform: [ubuntu-latest, macos-latest]
```
Comment (neolit123) [inline .github/workflows/test.yml:8]: do we care about `windows` too here?
i'd probably use `ubuntu-18.04, macos, windows`
the macos and windows don't seem to have versions as part of the names.

Comment (dims) [reply .github/workflows/test.yml:8]: Let's do windows in a subsequent patch @neolit123 i kicked the tires and did not get a clean run. Here's a clean run with the changes in this PR:
https://github.com/dims/klog/actions/runs/73799189

( windows failures that we can chase later - https://github.com/dims/klog/actions/runs/73797937 )

Comment (dims) [reply .github/workflows/test.yml:8]: let's pin if we see a problem with -latest please

**Comment by neolit123** [comment]
/lgtm

i can log a couple of issues:
- remove travis.
- add windows in the workflows and debug why it fails.

**Pull request merged**